vcpkg/users/binarycaching-troubleshooting.md

```diff
@@ -0,0 +1,286 @@
+
+### Cause 1: Vcpkg lacks write permissions to the binary cache
+<!--
+Steps to reproduce:
+1. Create a NuGet feed
+2. $env:VCPKG_BINARY_SOURCES="clear;nuget,C:/nugetfeed,read"
+3. Run ./vcpkg install rapidjson
+-->
+You are missing the following message in your output.
+```
+Uploading binaries for 'rapidjson:x64-windows' to <binary source> source <url>.
+Stored binaries in 1 destinations in 1.5 s.
+```
+
+Vcpkg skipped uploading the binary package to your binary cache.
+* Ensure your [binary cache configuration](binarycaching.md#configuration-syntax) is set to `write` or `readwrite`
+
+## Libraries rebuild instead of using remote binary cache
+
+[!INCLUDE [debug](../../includes/debug.md)]
+
+Your system rebuilds libraries locally, even when the required binary package is available in the remote binary cache.
+
+There is the absence of the following output:
+```
+Restored 1 package(s) from <remote binary cache> in 1.1 s. Use --debug to see more details.
+```
+
+### Cause 1: Vcpkg lacks read permissions from the remote binary cache
+
+Vcpkg choose to read your default binary cache over the remote one.
+* Ensure your [binary cache configuration](binarycaching.md#configuration-syntax) is set to `read` or `readwrite`
+
+### Cause 2: The remote binary cache is empty
+
+The remote cache should contain a list of binary packages you've pushed. 
+* Refer to the [empty binary cache](#empty-cache) section.
+
+### Cause 3: Differences between local and remote build environments
+
+Each package in binary cache is labeled with an [ABI hash](./binarycaching.md#abi-hash) which contains compiler versions, sources, and other information to distinguish between binary packages. If the locally computed ABI hash doesn't match the one stored remotely, the package isn't retrieved.
+* Consult the [ABI Hash mismatch troubleshooting guide](#abi-mismatch) to determine the root cause.
+
+## Unexpected or frequent library rebuilds
+
+[!INCLUDE [debug](../../includes/debug.md)]
+
+In an unchanged environment and without updating vcpkg, you still find yourself rebuilding libraries occassionally.
+
+### Cause 1: Undetected changes in the build environment
+
+Each package in binary cache is labeled with an [ABI hash](./binarycaching.md#abi-hash) which contains compiler versions, sources, and other information to distinguish between binary packages. If the locally computed ABI hash doesn't match the one stored remotely, the package isn't retrieved.
+* Consult the [ABI Hash mismatch troubleshooting guide](#abi-mismatch) to determine the root cause.
+
+## <a name="abi-mismatch"></a> Troubleshooting ABI hash mismatch
+
+[!INCLUDE [debug](../../includes/debug.md)]
+
+This guide is intended for users to diagnose why they have different ABI hashes for two identically named binary packages.
+
+### Comparing two binary packages
```

Comment:
```suggestion
```

Reply:
I'm in favor of keeping this to separate the sections. @vicroms thoughts?